dlc-manager/src/utils.rs

```diff
@@ -83,7 +86,30 @@ where
     // Add base cost of fund tx + CET / 2 and a CET output to the collateral.
     let appr_required_amount =
         own_collateral + get_half_common_fee(fee_rate)? + dlc::util::weight_to_fee(124, fee_rate)?;
-    let utxos = wallet.get_utxos_for_amount(appr_required_amount, fee_rate, true)?;
+    let mut utxos =
+        wallet.get_utxos_for_amount(appr_required_amount, fee_rate, true, &change_spk)?;
+    let total_value: u64 = utxos.iter().map(|x| x.tx_out.value).sum();
+    let min_change_value = change_addr.script_pubkey().dust_value().to_sat();
+    let (inputs_weight, change_weight) = get_inputs_and_change_weight(
+        &utxos
+            .iter()
+            .map(|x| (x.tx_out.script_pubkey.as_ref(), 107))
+            .collect::<Vec<_>>(),
+        &change_spk,
+    )?;
+    let inputs_fee = weight_to_fee(inputs_weight, fee_rate)?;
+    let change_fee = weight_to_fee(change_weight, fee_rate)?;
+    // We need to have a change output, if we didn't on first try, we request an amount which
+    // includes minimum value for the change output as well as the fee for it.
+    if total_value < appr_required_amount + min_change_value + inputs_fee + change_fee {
```

Comment:
So your proposal is just to error if there is not enough for change?

Reply:
I think that's okay, yeah. I know we have dealt with stuff like this, but I can't really blame `rust-dlc` if the API is clear and working as intended.

Reply:
Updated to return an error.

Reply:
To be honest I'm still pondering on this. Wouldn't it be easier to just request an amount that includes a change output + minimum spendable amount for it from the beginning?

Reply:
I think that's reasonable too and it seems simpler than what we are currently doing. I remember thinking that it might be a good idea to give more control to consumers of this API, but I can't figure out why now.